src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -1669,21 +1674,36 @@ public List<Action> retrieveActions(
     }
   }
 
+  private static LoadingCache<String, Boolean> createPrivateModeCache() {
+    // TODO configurable duration Potentially using Duration.parse.abs?
+    return Caffeine.newBuilder()
+        .expireAfterWrite(Duration.ofHours(20))
```

Comment:
Now that we know this works, maybe make the cache expiry configurable via `SystemProperties`?